packages/main/src/TabContainer.js

```diff
@@ -289,17 +326,25 @@ class TabContainer extends UI5Element {
 
 		this._handleResize = this._handleResize.bind(this);
 
-		// Init ScrollEnablement
-		this._scrollEnablement = new ScrollEnablement(this);
-		this._scrollEnablement.attachEvent("scroll", this._updateScrolling.bind(this));
-
 		// Init ItemNavigation
 		this._itemNavigation = new ItemNavigation(this, {
 			getItemsCallback: () => this._getTabs(),
 		});
 	}
 
 	onBeforeRendering() {
+		// update selected tab
+		const tabs = this._getTabs();
+		if (tabs.length) {
+			const selectedTabs = tabs.filter(tab => tab.selected);
+			if (selectedTabs.length) {
+				this._selectedTab = selectedTabs[0];
+			} else {
+				this._selectedTab = tabs[0];
+				this._selectedTab.selected = true;
```

Comment:
This is an antipattern. You must not set public properties to the component or its children. The app may set the correct "selected" value 1 second later, f.e. after a database call has been made, and to a different tab. Yet, you set it internally to the first, which may break the logic of execution later. Also, most apps will use some programming model to set public state, and when you do this, it gets desynced from the model. The app does not know that you changed a public property. Instead, you must use implied "selected" property of the tab, if this is really needed. We usually call these implied values `_effectiveSomething`.